src/serialization/preprocessor.cpp

```diff
@@ -1152,8 +1153,10 @@ bool preprocessor_data::get_chunk()
 
 			std::string symbol = items.front();
 			items.erase(items.begin());
-			int found_arg = 0, found_enddef = 0;
-			std::string buffer;
+			int found_arg = 0, found_enddef = 0, found_deprecate = 0;
+			int deprecation_level = -1;
```

Comment:
I am trying out a new analysis tool. It reports:

There are two other constructors. They do not initialize `deprecation_level`

Reply:
I'm confused... there's no constructor in sight here?

Reply:
We're inside one of the three. This one sets deprecation_level to -1 the other two leave it uninitialized.

Reply:
Uh, this is a local variable in `preprocessor_data::get_chunk`...

Reply:
> Found 2 possible new issues  
> 
> src/serialization/preprocessor.hpp
> 
>     Member variable 'preproc_define::deprecation_level' is not initialized in the
>     constructor.
>         35 preproc_define()
> 
>     Member variable 'preproc_define::deprecation_level' is not initialized in the
>     constructor.
>         45 explicit preproc_define(const std::string& val)

Reply:
Ahhh, I see it now. I was confused by the location you chose for this comment; the actual error lies in preprocessor.hpp.

Reply:
For the record, this issue should be fixed now as well.